extensions/wikia/Chat2/Chat.class.php

```diff
@@ -40,7 +39,7 @@ static public function echoCookies(){
 		if( !$wgUser->isLoggedIn() ) {
 			return array("key" => false ) ;
 		}
-		$key = md5( $wgUser->getId() . "_" . time() . '_' .  mt_rand(0, 65535) );
+		$key = sha1( $wgUser->getId() . "_" . microtime() . '_' .  mt_rand() );
 		$wgMemc->set($key, array( "user_id" => $wgUser->getId(), "cookie" => $_COOKIE) , 60*60*48);
```

Comment:
@SeanColombo: as we're here, please change the structure of the key to be not just the hash, but to contain some sort of a prefix, eg. `Chat::cookies::<hash>`